packages/smoldot-test-utils/src/index.ts

```diff
@@ -1,69 +1,152 @@
-  const id = JSON.parse(requestJSON).id;
-  return systemHealthReponse(id, 1);
+  const { id } = JSON.parse(requestJSON) as JsonRpcObject;
+  return systemHealthReponse(id, { isSyncing: false, peerCount: 1, shouldHavePeers: true });
 }
 
-// Always responds to a health request with a response that has no peers
-const unhealthyResponder = (requestJSON: string) => {
-  const id = JSON.parse(requestJSON).id;
-  return systemHealthReponse(id, 0);
+/**
+ * devChainHealthResponder - takes a JSON request that is expected to look like
+ * a valid "system_health" RPC request and returns a valid matching healthy
+ * response for a "dev chain". It takes the ID from the request and ensures the
+ * response has the same ID.
+ *
+ * @param requestJSON - the RPC request JSON
+ * @returns a JSON string
+ */
+export const devChainHealthResponder = (requestJSON: string): string => {
+  const { id } = JSON.parse(requestJSON) as JsonRpcObject;
+  return devChainHealthResponse(id);
 }
 
-// Always responds with an error reponse
-export const erroringResponder = (requestJSON: string) => {
-  const id = JSON.parse(requestJSON).id;
+/**
+ * unhealthyResponder - takes a JSON request that is expected to look like any
+ * valid RPC request and and always generates an RPC error response.
+ *
+ * @param requestJSON - the RPC request JSON
+ * @returns a JSON string
+ */
+export const erroringResponder = (requestJSON: string): string => {
+  const { id } = JSON.parse(requestJSON) as JsonRpcObject;
   return `{ "id": ${id}, "jsonrpc": "2.0", "error": {"code": 666, "message": "boom!" } }`;
 }
 
+
+/**
+ * Orchestrators - these are higher order factory functions take the
+ * definitions of a sequence of reponses to respond with and returns
+ * `RpcResponder` function that when called will return a reponse matching what
+ * was supplied one-by-one
+ */
+
+const NO_MORE_RESPONSES = "json_rpc_callback was called but there are no mock responses left to respond with";
+
 // Orchestrates a sequence of has peers / has no peers responses
-export const customHealthResponder = (hasPeers: boolean[]) => {
-  return (requestJSON: string) => {
-    if (hasPeers.shift()) {
-      return healthyResponder(requestJSON);
-    } else {
-      return unhealthyResponder(requestJSON);
+export const customHealthResponder = (healthResponses: SystemHealth[]) => {
+  return (requestJSON: string): string => {
+    const { id } = JSON.parse(requestJSON) as JsonRpcObject;
+    const health = healthResponses.shift();
+    if (!health) { 
+      throw new Error(NO_MORE_RESPONSES); 
     }
+
+    return systemHealthReponse(id, health);
   };
 }
 
-// dev chains never have peers
-const devChainHealthResponse = (id: number) => {
-  return `{"jsonrpc":"2.0","id":${id} ,"result":{"isSyncing":true,"peers":0,"shouldHavePeers":false}}`;
+// Orchestrates an `RpcResponder` with a queue of mock reponses to respond with.
+export const respondWith = (jsonResponses: string[]) => {
+  return (_: string): string => {
```

Comment:
Following up our conversation yesterday,  I am not sure why there is a need for this `return (_: string): string` to exist at all. Couldn't that be: `(): string` ? 

Not a blocker ofc.

Reply:
I configured the rule to allow underscores as ignored param names